### PR TITLE
DOC-424 DOC-425 Snowflake Insights Guide

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Adds a guide for ingesting Snowflake usage info into Insights.

## Changelog [New | Bug | Docs]

NOCHANGELOG

### CODE REVIEW

**Comment by hellendag** [comment]
Kicked off a manual run of the docs revamp GH action to test the change I merged earlier.

**Review by erinkcochran87 — APPROVED** [review]
I'm not sure what to do about the repeated `create_snowflake_insights_asset_and_schedule` content, but I don't think it needs to be a blocker for now. Otherwise left a few comments about some copy pasta and nits.

File: docs/docs-beta/docs/dagster-plus/insights/snowflake.md [hunk 1]
```diff
@@ -1,7 +1,84 @@
...
+<Tabs>
+  <TabItem value="before" label="Before">
+    <CodeExample filePath="dagster-plus/insights/snowflake/dbt_project.yml" language="python" />
+  </TabItem>
+  <TabItem value="after" label="After" default>
+    <CodeExample filePath="dagster-plus/insights/snowflake/dbt_project_insights.yml" language="python" />
+  </TabItem>
+</Tabs>
+
+This adds a comment to each query, which is used by Dagster+ to attribute cost metrics to the correct assets.
```
[the remark this answers is not in the document]
Comment (OwenKephart) [reply docs/docs-beta/docs/dagster-plus/insights/snowflake.md:84]: Tucked it into the tab item!

File: docs/docs-beta/docs/dagster-plus/insights/snowflake.md [hunk 2]
```diff
@@ -1,7 +1,84 @@
...
+  </TabItem>
+</Tabs>
+
+## Tracking usage with dagster-dbt
+
+If you use `dagster-dbt` to manage a dbt project that targets Snowflake, you can emit usage metrics to the Dagster+ API with the `DbtCliResource`.
+
+First, add a `.with_insights()` call to your `dbt.cli()` command(s), and add Snowflake-specific insights definitions to your code using `create_snowflake_insights_asset_and_schedule`.
+
+These additional definitions are required because Snowflake usage information is only available after a delay. These definitions automatically handle running a computation on a schedule to ingest Snowflake usage information from the previous hour.
```
[the remark this answers is not in the document]
Comment (OwenKephart) [reply docs/docs-beta/docs/dagster-plus/insights/snowflake.md:58]: Yeah I wasn't a huge fan of this -- my decision making process was basically that I wanted each section to be independently consumable (i.e. you find yourself in the dbt section, you don't need to go anywhere else), but agreed it's kinda annoying

File: docs/docs-beta/docs/dagster-plus/insights/snowflake.md [hunk 3]
```diff
@@ -1,7 +1,84 @@
...
+
+## Tracking usage with the SnowflakeResource
+
+The `dagster-cloud` package provides an `InsightsSnowflakeResource`, which is a drop-in replacement for the `SnowflakeResource` provided by `dagster-snowflake`.
+
+This resource will emit BigQuery usage metrics to the Dagster+ Insights API whenever it makes a query.
+
+To enable this behavior, replace usage of `SnowflakeResource` with `InsightsSnowflakeResource`, and add Snowflake-specific insights definitions to your code using `create_snowflake_insights_asset_and_schedule`.
+
+These additional definitions are required because Snowflake usage information is only available after a delay. These definitions automatically handle running a computation on a schedule to ingest Snowflake usage information from the previous hour.
```
[the remark this answers is not in the document]
Comment (OwenKephart) [reply docs/docs-beta/docs/dagster-plus/insights/snowflake.md:37]: Yeah happy for ideas here -- it's honestly not the greatest user experience at the moment and I thought it deserved some explanation as to why you had to add all this random code

**Comment by OwenKephart** [comment]
### Merge activity

* **Aug 29, 4:40 PM EDT**: @OwenKephart started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/24047).

**Pull request merged**